### ISSUE TITLE
Driver won't start on 32bit OS

### ISSUE THREAD

**Issue by TitovLab** [OpenVPN/ovpn-dco-win#43]
Hi! Driver didn't start on clean Windows 10 **32bit** OS:
![image](https://user-images.githubusercontent.com/11898568/228513564-36a77c34-9b08-4460-bd15-700a44334c3d.png)

Tried on one clean physical machine (Win 10 22H2) and two clean Hyper-V VM's: (Win 10 21H2 and 22H2).
No third party antivirus or software, clean install.
Also tried different versions:
OpenVPN-2.6.2-I001-x86.msi
OpenVPN-2.6.1-I001-x86.msi
OpenVPN-2.6.0-I005-x86.msi

On 64bit OS all fine, error only on 32bit OS.

C:\Windows\INF\setupapi.dev.log
```
[Device Install Log]
     OS Version = 10.0.19045
     Service Pack = 0.0
     Suite = 0x0100
     ProductType = 1
     Architecture = x86

[BeginLog]

[Boot Session: 2023/03/28 11:45:32.630]

>>>  [Setup Import Driver Package - C:\Windows\system32\spool\tools\Microsoft Print To PDF\prnms009.Inf]
>>>  Section start 2023/03/28 11:46:22.937
      cmd: C:\Windows\System32\spoolsv.exe
     inf: Provider: Microsoft
     inf: Class GUID: {4D36E979-E325-11CE-BFC1-08002BE10318}
     inf: Driver Version: 06/21/2006,10.0.19041.1806
     inf: Catalog File: prnms009.cat
     ump: Import flags: 0x0000000D
     pol: {Driver package policy check} 11:46:23.161
     pol: {Driver package policy check - exit(0x00000000)} 11:46:23.171
     sto: {Stage Driver Package: C:\Windows\system32\spool\tools\Microsoft Print To PDF\prnms009.Inf} 11:46:23.171
     inf:      {Query Configurability: C:\Windows\system32\spool\tools\Microsoft Print To PDF\prnms009.Inf} 11:46:23.203
     inf:           Driver package 'prnms009.Inf' is configurable.
     inf:      {Query Configurability: exit(0x00000000)} 11:46:23.203
     flq:      {FILE_QUEUE_COMMIT} 11:46:23.235
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft Print To PDF\MPDW_devmode_map.xml' to 'C:\Windows\System32\DriverStore\Temp\{2b785250-6227-a744-bc04-c095932b4d45}\MPDW_devmode_map.xml'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft Print To PDF\prnms009.cat' to 'C:\Windows\System32\DriverStore\Temp\{2b785250-6227-a744-bc04-c095932b4d45}\prnms009.cat'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft Print To PDF\prnms009.Inf' to 'C:\Windows\System32\DriverStore\Temp\{2b785250-6227-a744-bc04-c095932b4d45}\prnms009.Inf'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft Print To PDF\MPDW-PDC.xml' to 'C:\Windows\System32\DriverStore\Temp\{2b785250-6227-a744-bc04-c095932b4d45}\MPDW-PDC.xml'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft Print To PDF\MPDW-manifest.ini' to 'C:\Windows\System32\DriverStore\Temp\{2b785250-6227-a744-bc04-c095932b4d45}\MPDW-manifest.ini'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft Print To PDF\MPDW-pipelineconfig.xml' to 'C:\Windows\System32\DriverStore\Temp\{2b785250-6227-a744-bc04-c095932b4d45}\MPDW-pipelineconfig.xml'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft Print To PDF\MPDW-constraints.js' to 'C:\Windows\System32\DriverStore\Temp\{2b785250-6227-a744-bc04-c095932b4d45}\MPDW-constraints.js'.
     flq:      {FILE_QUEUE_COMMIT - exit(0x00000000)} 11:46:23.412
     sto:      {DRIVERSTORE IMPORT VALIDATE} 11:46:23.421
     sig:           Driver package catalog is valid.
     sig:           {_VERIFY_FILE_SIGNATURE} 11:46:23.515
     sig:                Key      = prnms009.Inf
     sig:                FilePath = C:\Windows\System32\DriverStore\Temp\{2b785250-6227-a744-bc04-c095932b4d45}\prnms009.Inf
     sig:                Catalog  = C:\Windows\System32\DriverStore\Temp\{2b785250-6227-a744-bc04-c095932b4d45}\prnms009.cat
     sig:                Success: File is signed in catalog.
     sig:           {_VERIFY_FILE_SIGNATURE exit(0x00000000)} 11:46:23.547
     sto:      {DRIVERSTORE IMPORT VALIDATE: exit(0x00000000)} 11:46:23.578
     sig:      Signer Score  = 0x0D000003 (Inbox)
     sig:      Signer Name   = Microsoft Windows
     sto:      {Core Driver Package Import: prnms009.inf_x86_93dc61f7a147d775} 11:46:23.593
     sto:           {DRIVERSTORE IMPORT BEGIN} 11:46:23.593
     sto:           {DRIVERSTORE IMPORT BEGIN: exit(0x00000000)} 11:46:23.593
     cpy:           {Copy Directory: C:\Windows\System32\DriverStore\Temp\{2b785250-6227-a744-bc04-c095932b4d45}} 11:46:23.593
     cpy:                Target Path = C:\Windows\System32\DriverStore\FileRepository\prnms009.inf_x86_93dc61f7a147d775
     cpy:           {Copy Directory: exit(0x00000000)} 11:46:23.656
     idb:           {Register Driver Package: C:\Windows\System32\DriverStore\FileRepository\prnms009.inf_x86_93dc61f7a147d775\prnms009.Inf} 11:46:23.671
     idb:                Created driver package object 'prnms009.inf_x86_93dc61f7a147d775' in DRIVERS database node.
     idb:                Created driver INF file object 'oem0.inf' in DRIVERS database node.
     idb:                Registered driver package 'prnms009.inf_x86_93dc61f7a147d775' with 'oem0.inf'.
     idb:           {Register Driver Package: exit(0x00000000)} 11:46:23.687
     idb:           {Publish Driver Package: C:\Windows\System32\DriverStore\FileRepository\prnms009.inf_x86_93dc61f7a147d775\prnms009.Inf} 11:46:23.687
     idb:                Activating driver package 'prnms009.inf_x86_93dc61f7a147d775'.
     cpy:                Published 'prnms009.inf_x86_93dc61f7a147d775\prnms009.inf' to 'oem0.inf'.
     idb:                Indexed 2 device IDs for 'prnms009.inf_x86_93dc61f7a147d775'.
     sto:                Flushed driver database node 'DRIVERS'. Time = 15 ms
     sto:                Flushed driver database node 'SYSTEM'. Time = 94 ms
     idb:           {Publish Driver Package: exit(0x00000000)} 11:46:23.828
     sto:           {DRIVERSTORE IMPORT END} 11:46:23.843
     dvi:                Flushed all driver package files to disk. Time = 31 ms
     sig:                Installed catalog 'prnms009.cat' as 'oem0.cat'.
     sto:           {DRIVERSTORE IMPORT END: exit(0x00000000)} 11:46:23.906
     sto:      {Core Driver Package Import: exit(0x00000000)} 11:46:23.906
     sto: {Stage Driver Package: exit(0x00000000)} 11:46:23.921
<<<  Section end 2023/03/28 11:46:23.999
<<<  [Exit status: SUCCESS]


>>>  [Setup Import Driver Package - C:\Windows\system32\spool\tools\Microsoft XPS Document Writer\prnms001.Inf]
>>>  Section start 2023/03/28 11:46:25.345
      cmd: C:\Windows\System32\spoolsv.exe
     inf: Provider: Microsoft
     inf: Class GUID: {4D36E979-E325-11CE-BFC1-08002BE10318}
     inf: Driver Version: 06/21/2006,10.0.19041.1806
     inf: Catalog File: prnms001.cat
     ump: Import flags: 0x0000000D
     pol: {Driver package policy check} 11:46:25.390
     pol: {Driver package policy check - exit(0x00000000)} 11:46:25.390
     sto: {Stage Driver Package: C:\Windows\system32\spool\tools\Microsoft XPS Document Writer\prnms001.Inf} 11:46:25.390
     inf:      {Query Configurability: C:\Windows\system32\spool\tools\Microsoft XPS Document Writer\prnms001.Inf} 11:46:25.423
     inf:           Driver package 'prnms001.Inf' is configurable.
     inf:      {Query Configurability: exit(0x00000000)} 11:46:25.423
     flq:      {FILE_QUEUE_COMMIT} 11:46:25.423
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft XPS Document Writer\prnms001.cat' to 'C:\Windows\System32\DriverStore\Temp\{f6fd8423-81e9-7b45-8325-83acd2ebde5d}\prnms001.cat'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft XPS Document Writer\prnms001.Inf' to 'C:\Windows\System32\DriverStore\Temp\{f6fd8423-81e9-7b45-8325-83acd2ebde5d}\prnms001.Inf'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft XPS Document Writer\mxdwdui.dll' to 'C:\Windows\System32\DriverStore\Temp\{f6fd8423-81e9-7b45-8325-83acd2ebde5d}\mxdwdui.dll'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft XPS Document Writer\MXDW-manifest.ini' to 'C:\Windows\System32\DriverStore\Temp\{f6fd8423-81e9-7b45-8325-83acd2ebde5d}\MXDW-manifest.ini'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft XPS Document Writer\MXDW.gpd' to 'C:\Windows\System32\DriverStore\Temp\{f6fd8423-81e9-7b45-8325-83acd2ebde5d}\MXDW.gpd'.
     flq:           Copying 'C:\Windows\system32\spool\tools\Microsoft XPS Document Writer\MXDW-pipelineconfig.xml' to 'C:\Windows\System32\DriverStore\Temp\{f6fd8423-81e9-7b45-8325-83acd2ebde5d}\MXDW-pipelineconfig.xml'.
     flq:      {FILE_QUEUE_COMMIT - exit(0x00000000)} 11:46:25.611
     sto:      {DRIVERSTORE IMPORT VALIDATE} 11:46:25.611
     sig:           Driver package catalog is valid.
     sig:           {_VERIFY_FILE_SIGNATURE} 11:46:25.765
     sig:                Key      = prnms001.Inf
     sig:                FilePath = C:\Windows\System32\DriverStore\Temp\{f6fd8423-81e9-7b45-8325-83acd2ebde5d}\prnms001.Inf
     sig:                Catalog  = C:\Windows\System32\DriverStore\Temp\{f6fd8423-81e9-7b45-8325-83acd2ebde5d}\prnms001.cat
     sig:                Success: File is signed in catalog.
     sig:           {_VERIFY_FILE_SIGNATURE exit(0x00000000)} 11:46:25.812
     sto:      {DRIVERSTORE IMPORT VALIDATE: exit(0x00000000)} 11:46:25.843
     sig:      Signer Score  = 0x0D000003 (Inbox)
     sig:      Signer Name   = Microsoft Windows
     sto:      {Core Driver Package Import: prnms001.inf_x86_612b3994b58c1451} 11:46:25.843
     sto:           {DRIVERSTORE IMPORT BEGIN} 11:46:25.843
     sto:           {DRIVERSTORE IMPORT BEGIN: exit(0x00000000)} 11:46:25.843
     cpy:           {Copy Directory: C:\Windows\System32\DriverStore\Temp\{f6fd8423-81e9-7b45-8325-83acd2ebde5d}} 11:46:25.843
     cpy:                Target Path = C:\Windows\System32\DriverStore\FileRepository\prnms001.inf_x86_612b3994b58c1451
     cpy:           {Copy Directory: exit(0x00000000)} 11:46:25.876
     idb:           {Register Driver Package: C:\Windows\System32\DriverStore\FileRepository\prnms001.inf_x86_612b3994b58c1451\prnms001.Inf} 11:46:25.876
     idb:                Created driver package object 'prnms001.inf_x86_612b3994b58c1451' in DRIVERS database node.
     idb:                Created driver INF file object 'oem1.inf' in DRIVERS database node.
     idb:                Registered driver package 'prnms001.inf_x86_612b3994b58c1451' with 'oem1.inf'.
     idb:           {Register Driver Package: exit(0x00000000)} 11:46:25.893
     idb:           {Publish Driver Package: C:\Windows\System32\DriverStore\FileRepository\prnms001.inf_x86_612b3994b58c1451\prnms001.Inf} 11:46:25.893
     idb:                Activating driver package 'prnms001.inf_x86_612b3994b58c1451'.
     cpy:                Published 'prnms001.inf_x86_612b3994b58c1451\prnms001.inf' to 'oem1.inf'.
     idb:                Indexed 2 device IDs for 'prnms001.inf_x86_612b3994b58c1451'.
     sto:                Flushed driver database node 'DRIVERS'. Time = 31 ms
     sto:                Flushed driver database node 'SYSTEM'. Time = 110 ms
     idb:           {Publish Driver Package: exit(0x00000000)} 11:46:26.094
     sto:           {DRIVERSTORE IMPORT END} 11:46:26.109
     dvi:                Flushed all driver package files to disk. Time = 32 ms
     sig:                Installed catalog 'prnms001.cat' as 'oem1.cat'.
     sto:           {DRIVERSTORE IMPORT END: exit(0x00000000)} 11:46:26.208
     sto:      {Core Driver Package Import: exit(0x00000000)} 11:46:26.208
     sto: {Stage Driver Package: exit(0x00000000)} 11:46:26.229
<<<  Section end 2023/03/28 11:46:26.531
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{572ACD02-F789-458B-ACD8-FD258BCB7917}]
>>>  Section start 2023/03/28 21:52:43.430
      cmd: C:\Windows\System32\spoolsv.exe
!    dvi: Access denied from Query and Remove
<<<  Section end 2023/03/28 21:52:43.446
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{D2CA61DB-EF3F-4D1C-83EC-17F071CC8BC7}]
>>>  Section start 2023/03/28 21:52:43.540
      cmd: C:\Windows\System32\spoolsv.exe
<<<  Section end 2023/03/28 21:52:43.555
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{47FF3959-BCD1-4802-B7B8-F53F060FEBED}]
>>>  Section start 2023/03/28 21:52:43.776
      cmd: C:\Windows\System32\spoolsv.exe
!    dvi: Access denied from Query and Remove
<<<  Section end 2023/03/28 21:52:43.807
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{FD4E60E1-C60A-43B8-BD39-4EE09A27C39D}]
>>>  Section start 2023/03/28 21:52:43.979
      cmd: C:\Windows\System32\spoolsv.exe
!    dvi: Access denied from Query and Remove
<<<  Section end 2023/03/28 21:52:44.010
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{A40CDDF2-E951-4382-B996-3DB207486694}]
>>>  Section start 2023/03/28 21:52:44.153
      cmd: C:\Windows\System32\spoolsv.exe
!    dvi: Access denied from Query and Remove
<<<  Section end 2023/03/28 21:52:44.182
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{4EBDCFF3-44D2-4489-9993-2F3376A3BA27}]
>>>  Section start 2023/03/28 21:52:44.338
      cmd: C:\Windows\System32\spoolsv.exe
!    dvi: Access denied from Query and Remove
<<<  Section end 2023/03/28 21:52:44.370
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{177BFFA1-7B75-44EE-9F3B-A4B6984AE2D9}]
>>>  Section start 2023/03/28 21:52:44.448
      cmd: C:\Windows\System32\spoolsv.exe
!    dvi: Access denied from Query and Remove
<<<  Section end 2023/03/28 21:52:44.479
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{8BB003AE-AF58-4181-AEF5-4120EFEE8D16}]
>>>  Section start 2023/03/28 21:52:44.557
      cmd: C:\Windows\System32\spoolsv.exe
!    dvi: Access denied from Query and Remove
<<<  Section end 2023/03/28 21:52:44.588
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{0392B9C0-D4D3-4823-8AD6-4D20101E7ADB}]
>>>  Section start 2023/03/28 21:52:44.666
      cmd: C:\Windows\System32\spoolsv.exe
!    dvi: Access denied from Query and Remove
<<<  Section end 2023/03/28 21:52:44.698
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{4825E1BF-FE29-4700-8420-520571599685}]
>>>  Section start 2023/03/28 21:52:44.776
      cmd: C:\Windows\System32\spoolsv.exe
<<<  Section end 2023/03/28 21:52:44.807
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{1F6BC8D8-C60D-4609-B340-0B4B199660EF}]
>>>  Section start 2023/03/28 21:52:44.885
      cmd: C:\Windows\System32\spoolsv.exe
<<<  Section end 2023/03/28 21:52:44.901
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{7108E7F5-ABD1-46BC-A8DE-A44838A2B288}]
>>>  Section start 2023/03/28 21:52:44.979
      cmd: C:\Windows\System32\spoolsv.exe
!    dvi: Access denied from Query and Remove
<<<  Section end 2023/03/28 21:52:44.995
<<<  [Exit status: SUCCESS]


>>>  [Delete Device - SWD\PRINTENUM\{673F437B-4BF1-4EFE-AF81-B13B287CDAAC}]
>>>  Section start 2023/03/28 21:52:45.073
      cmd: C:\Windows\System32\spoolsv.exe
!    dvi: Access denied from Query and Remove
<<<  Section end 2023/03/28 21:52:45.088
<<<  [Exit status: SUCCESS]


[Boot Session: 2023/03/29 13:33:03.315]

>>>  [SetupCopyOEMInf - C:\Program Files\Common Files\ovpn-dco\Win10\ovpn-dco.inf]
>>>  Section start 2023/03/29 13:36:36.333
      cmd: C:\Windows\system32\MsiExec.exe -Embedding 27E1C658BA76CB230AEE949DF4DD65D0 E Global\MSI0000
     inf: Copy style: 0x00000000
     sto: {Setup Import Driver Package: C:\Program Files\Common Files\ovpn-dco\Win10\ovpn-dco.inf} 13:36:36.364
     inf:      Provider: OpenVPN, Inc
     inf:      Class GUID: {4d36e972-e325-11ce-bfc1-08002be10318}
     inf:      Driver Version: 03/02/2023,0.9.2.0
     inf:      Catalog File: ovpn-dco.cat
     ump:      Import flags: 0x00000001
     pol:      {Driver package policy check} 13:36:36.442
     pol:      {Driver package policy check - exit(0x00000000)} 13:36:36.442
     sto:      {Stage Driver Package: C:\Program Files\Common Files\ovpn-dco\Win10\ovpn-dco.inf} 13:36:36.442
     inf:           {Query Configurability: C:\Program Files\Common Files\ovpn-dco\Win10\ovpn-dco.inf} 13:36:36.458
     inf:                Driver package uses WDF.
     inf:                Driver package 'ovpn-dco.inf' is configurable.
     inf:           {Query Configurability: exit(0x00000000)} 13:36:36.474
     flq:           {FILE_QUEUE_COMMIT} 13:36:36.474
     flq:                Copying 'C:\Program Files\Common Files\ovpn-dco\Win10\ovpn-dco.cat' to 'C:\Windows\System32\DriverStore\Temp\{fa86f26e-de3c-234c-bbc2-08672d131d0d}\ovpn-dco.cat'.
     flq:                Copying 'C:\Program Files\Common Files\ovpn-dco\Win10\ovpn-dco.inf' to 'C:\Windows\System32\DriverStore\Temp\{fa86f26e-de3c-234c-bbc2-08672d131d0d}\ovpn-dco.inf'.
     flq:                Copying 'C:\Program Files\Common Files\ovpn-dco\Win10\ovpn-dco.sys' to 'C:\Windows\System32\DriverStore\Temp\{fa86f26e-de3c-234c-bbc2-08672d131d0d}\ovpn-dco.sys'.
     flq:           {FILE_QUEUE_COMMIT - exit(0x00000000)} 13:36:36.552
     sto:           {DRIVERSTORE IMPORT VALIDATE} 13:36:36.552
     sig:                Driver package catalog is valid.
     sig:                {_VERIFY_FILE_SIGNATURE} 13:36:36.614
     sig:                     Key      = ovpn-dco.inf
     sig:                     FilePath = C:\Windows\System32\DriverStore\Temp\{fa86f26e-de3c-234c-bbc2-08672d131d0d}\ovpn-dco.inf
     sig:                     Catalog  = C:\Windows\System32\DriverStore\Temp\{fa86f26e-de3c-234c-bbc2-08672d131d0d}\ovpn-dco.cat
     sig:                     Success: File is signed in catalog.
     sig:                {_VERIFY_FILE_SIGNATURE exit(0x00000000)} 13:36:36.646
     sto:           {DRIVERSTORE IMPORT VALIDATE: exit(0x00000000)} 13:36:36.646
     sig:           Signer Score  = 0x0D000005 (WHQL)
     sig:           Signer Name   = Microsoft Windows Hardware Compatibility Publisher
     sig:           Submission ID = 53719220_13541212701107156_1152921505696008664
     sto:           {Core Driver Package Import: ovpn-dco.inf_x86_555c25041c7ab5fc} 13:36:36.661
     sto:                {DRIVERSTORE IMPORT BEGIN} 13:36:36.661
     sto:                {DRIVERSTORE IMPORT BEGIN: exit(0x00000000)} 13:36:36.661
     cpy:                {Copy Directory: C:\Windows\System32\DriverStore\Temp\{fa86f26e-de3c-234c-bbc2-08672d131d0d}} 13:36:36.661
     cpy:                     Target Path = C:\Windows\System32\DriverStore\FileRepository\ovpn-dco.inf_x86_555c25041c7ab5fc
     cpy:                {Copy Directory: exit(0x00000000)} 13:36:36.661
     idb:                {Register Driver Package: C:\Windows\System32\DriverStore\FileRepository\ovpn-dco.inf_x86_555c25041c7ab5fc\ovpn-dco.inf} 13:36:36.661
     idb:                     Created driver package object 'ovpn-dco.inf_x86_555c25041c7ab5fc' in DRIVERS database node.
     idb:                     Created driver INF file object 'oem2.inf' in DRIVERS database node.
     idb:                     Registered driver package 'ovpn-dco.inf_x86_555c25041c7ab5fc' with 'oem2.inf'.
     idb:                {Register Driver Package: exit(0x00000000)} 13:36:36.677
     idb:                {Publish Driver Package: C:\Windows\System32\DriverStore\FileRepository\ovpn-dco.inf_x86_555c25041c7ab5fc\ovpn-dco.inf} 13:36:36.692
     idb:                     Activating driver package 'ovpn-dco.inf_x86_555c25041c7ab5fc'.
     cpy:                     Published 'ovpn-dco.inf_x86_555c25041c7ab5fc\ovpn-dco.inf' to 'oem2.inf'.
     idb:                     Indexed 2 device IDs for 'ovpn-dco.inf_x86_555c25041c7ab5fc'.
     sto:                     Flushed driver database node 'DRIVERS'. Time = 15 ms
     sto:                     Flushed driver database node 'SYSTEM'. Time = 32 ms
     idb:                {Publish Driver Package: exit(0x00000000)} 13:36:36.755
     sto:                {DRIVERSTORE IMPORT END} 13:36:36.755
     dvi:                     Flushed all driver package files to disk. Time = 15 ms
     sig:                     Installed catalog 'ovpn-dco.cat' as 'oem2.cat'.
     sto:                {DRIVERSTORE IMPORT END: exit(0x00000000)} 13:36:36.786
     sto:           {Core Driver Package Import: exit(0x00000000)} 13:36:36.786
     sto:      {Stage Driver Package: exit(0x00000000)} 13:36:36.786
     sto: {Setup Import Driver Package - exit (0x00000000)} 13:36:36.864
     inf: Driver Store Path: C:\Windows\System32\DriverStore\FileRepository\ovpn-dco.inf_x86_555c25041c7ab5fc\ovpn-dco.inf
     inf: Published Inf Path: C:\Windows\INF\oem2.inf
<<<  Section end 2023/03/29 13:36:36.880
<<<  [Exit status: SUCCESS]


>>>  [Device Install (UpdateDriverForPlugAndPlayDevices) - ovpn-dco]
>>>  Section start 2023/03/29 13:36:36.880
      cmd: C:\Windows\system32\MsiExec.exe -Embedding 27E1C658BA76CB230AEE949DF4DD65D0 E Global\MSI0000
     ndv: INF path: C:\Program Files\Common Files\ovpn-dco\Win10\ovpn-dco.inf
     ndv: Install flags: 0x00000005
!    ndv: Unable to find any matching devices.
<<<  Section end 2023/03/29 13:36:36.896
<<<  [Exit status: FAILURE(0xe000020b)]


>>>  [Device Install (DiInstallDevice) - ROOT\NET\0000]
>>>  Section start 2023/03/29 13:36:37.177
      cmd: C:\Windows\system32\MsiExec.exe -Embedding 27E1C658BA76CB230AEE949DF4DD65D0 E Global\MSI0000
     ndv: Flags: 0x00000000
     ndv: Search options: 0x00000002
     ndv: Searching default INF path
     dvi: {Build Driver List} 13:36:37.192
     dvi:      Searching for hardware ID(s):
     dvi:           ovpn-dco
     dvi:      Created Driver Node:
     dvi:           HardwareID   - ovpn-dco
     dvi:           InfName      - C:\Windows\System32\DriverStore\FileRepository\ovpn-dco.inf_x86_555c25041c7ab5fc\ovpn-dco.inf
     dvi:           DevDesc      - OpenVPN Data Channel Offload
     dvi:           Section      - ovpn-dco_Device.NT
     dvi:           Rank         - 0x00ff0000
     dvi:           Signer Score - WHQL
     dvi:           DrvDate      - 03/02/2023
     dvi:           Version      - 0.9.2.0
     dvi: {Build Driver List - exit(0x00000000)} 13:36:37.489
     dvi: {DIF_SELECTBESTCOMPATDRV} 13:36:37.489
     dvi:      Default installer: Enter 13:36:37.489
     dvi:           {Select Best Driver}
     dvi:                Class GUID of device changed to: {4d36e972-e325-11ce-bfc1-08002be10318}.
     dvi:                Selected Driver:
     dvi:                     Description - OpenVPN Data Channel Offload
     dvi:                     InfFile     - c:\windows\system32\driverstore\filerepository\ovpn-dco.inf_x86_555c25041c7ab5fc\ovpn-dco.inf
     dvi:                     Section     - ovpn-dco_Device
     dvi:           {Select Best Driver - exit(0x00000000)}
     dvi:      Default installer: Exit
     dvi: {DIF_SELECTBESTCOMPATDRV - exit(0x00000000)} 13:36:37.505
     ndv: Driver package 'C:\Windows\System32\DriverStore\FileRepository\ovpn-dco.inf_x86_555c25041c7ab5fc\ovpn-dco.inf' is already imported.
     sto: {Setup Import Driver Package: C:\Windows\INF\oem2.inf} 13:36:37.520
     sto:      Driver package already imported as 'oem2.inf'.
     sto: {Setup Import Driver Package - exit (0x00000000)} 13:36:37.520
     dvi: Searching for hardware ID(s):
     dvi:      ovpn-dco
     dvi: Class GUID of device changed to: {4d36e972-e325-11ce-bfc1-08002be10318}.
     ump: {Plug and Play Service: Device Install for ROOT\NET\0000}
     dvi:      {Core Device Install} 13:36:37.583
     dvi:           {Configure Device - ROOT\NET\0000} 13:36:37.583
     dvi:                Device Status: 0x01802001
     dvi:                Parent Device: HTREE\ROOT\0
     sto:                {Configure Driver Package: C:\Windows\System32\DriverStore\FileRepository\ovpn-dco.inf_x86_555c25041c7ab5fc\ovpn-dco.inf}
     sto:                     Source Filter  = ovpn-dco
     inf:                     Class GUID     = {4d36e972-e325-11ce-bfc1-08002be10318}
     inf:                     Class Options  = Configurable
     inf:                     {Configure Driver: OpenVPN Data Channel Offload}
     inf:                          Section Name = ovpn-dco_Device.NT
     inf:                          {Add Service: ovpn-dco}
     inf:                               Start Type    = 3
     inf:                               Service Type  = 1
     inf:                               Error Control = 1
     inf:                               Image Path    = \SystemRoot\System32\drivers\ovpn-dco.sys
     inf:                               Display Name  = ovpn-dco
     inf:                               Created new service 'ovpn-dco'.
     inf:                          {Add Service: exit(0x00000000)}
     inf:                          Hardware Id  = ovpn-dco
     inf:                          {Configure Driver Configuration: ovpn-dco_Device.NT}
     inf:                               Service Name  = ovpn-dco
     inf:                               Config Flags  = 0x00000000
     inf:                          {Configure Driver Configuration: exit(0x00000000)}
     inf:                     {Configure Driver: exit(0x00000000)}
     flq:                     {FILE_QUEUE_COMMIT} 13:36:37.646
     flq:                          Copying 'C:\Windows\System32\DriverStore\FileRepository\ovpn-dco.inf_x86_555c25041c7ab5fc\ovpn-dco.sys' to 'C:\Windows\System32\drivers\ovpn-dco.sys'.
     flq:                     {FILE_QUEUE_COMMIT - exit(0x00000000)} 13:36:37.661
     dvi:                     Existing files modified, may need to restart related services.
     sto:                {Configure Driver Package: exit(0x00000bc3)}
     dvi:                Restart required for any devices using this driver.
     dvi:                Install Device: Configuring device. 13:36:37.661
     dvi:                     Configuration: oem2.inf:ovpn-dco,ovpn-dco_Device
     dvi:                Install Device: Configuring device completed. 13:36:37.677
     dvi:                Device Status: 0x01802001
     dvi:                {Restarting Devices} 13:36:37.677
     dvi:                     Start: ROOT\NET\0000
!    dvi:                     Device 'ROOT\NET\0000' pending start: Device has problem: 0x38 (CM_PROB_NEED_CLASS_CONFIG), problem status: 0x00000000.
     dvi:                {Restarting Devices exit} 13:36:37.724
     dvi:           {Configure Device - exit(0x00000000)} 13:36:37.724
     dvi:      {Core Device Install - exit(0x00000000)} 13:36:37.724
     dvi:      Waiting for device post-install to complete. 13:36:37.740
     dvi:      Device post-install completed. 13:36:37.958
!    dvi:      Device Status: 0x01802401 [0x27 - 0xc0000034]
     ump: {Plug and Play Service: Device Install exit(00000000)}
<<<  Section end 2023/03/29 13:36:37.958
<<<  [Exit status: SUCCESS]
```

**Comment by lstipakov** [comment]
Thanks. 

My understanding is that `Device has problem: 0x38 (CM_PROB_NEED_CLASS_CONFIG), problem status: 0x00000000.` is not really an error but a [transient ](https://community.osr.com/discussion/289926/what-is-cm-prob-need-class-config) status. I have seen this on my machine, too.

I have to admit I never tested DCO driver on 32bit OS, although we do occasionally test OpenVPN installers on Win 7 32bit.

I'll look into it.

**Comment by lstipakov** [comment]
Close when new version is released.

**Comment by lstipakov** [comment]
Fixed in 2.6.3.
